x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcAssert.java

```diff
@@ -47,25 +47,31 @@
         }
     }
 
-    public static void assertResultSets(ResultSet expected, ResultSet actual) throws SQLException {
-        assertResultSets(expected, actual, null);
+    public static void assertResultSets(ResultSet expected, ResultSet actual, boolean lenientFloatingNumbers) throws SQLException {
+        assertResultSets(expected, actual, null, lenientFloatingNumbers);
     }
 
-    public static void assertResultSets(ResultSet expected, ResultSet actual, Logger logger) throws SQLException {
-        assertResultSets(expected, actual, logger, false);
+    public static void assertResultSets(ResultSet expected, ResultSet actual, Logger logger, boolean lenientFloatingNumbers) 
+           throws SQLException {
+        assertResultSets(expected, actual, logger, false, lenientFloatingNumbers);
     }
 
     /**
      * Assert the given result sets, potentially in a lenient way.
-     * When lenient is specified, the type comparison of a column is widden to reach a common, compatible ground.
+     * When lenientDataType is specified, the type comparison of a column is widden to reach a common, compatible ground.
      * This means promoting integer types to long and floating types to double and comparing their values.
-     * For example in a non-lenient, strict case a comparison between an int and a tinyint would fail, with lenient it will succeed as
-     * long as the actual value is the same.
+     * For example in a non-lenient, strict case a comparison between an int and a tinyint would fail, with lenientDataType it will succeed
+     * as long as the actual value is the same.
+     * Also, has the option of treating the numeric results for floating point numbers in a leninent way, if chosen to. Usually,
+     * we would want lening treatment for floating point numbers in sql-spec tests where the comparison is being made with H2.
+     * Typically, H2 doesn't return floating point numbers for some functions (AVG() being on example), and we'd want
+     * to be lenient about this, so that we can still compare the average values between H2 and ES-SQL.
      */
-    public static void assertResultSets(ResultSet expected, ResultSet actual, Logger logger, boolean lenient) throws SQLException {
+    public static void assertResultSets(ResultSet expected, ResultSet actual, Logger logger, boolean lenientDataType,
```

Comment:
Keep the previous method with reduced parameters and delegate to the new one with a default value. This would minimize the amount of changes since the leniency needs to be disabled/enabled just in some cases .

Reply:
Thanks @costin. I agree with the fact that there will be more changes to test methods, but this approach is intentional, so that the user of this method becomes aware of the leniency regarding floating point numbers, something that wasn't obvious before.

Reply:
Changed.